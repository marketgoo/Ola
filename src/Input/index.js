import React from 'react'
import { default as PT } from 'prop-types'
import cx from 'classnames'

const Input = React.forwardRef(({ className, error, type, ...props }, ref) => {
  const Element = type === 'textarea' ? 'textarea' : 'input'

  return <Element
    ref={ref}
    type={type}
    className={cx('ola_input', { 'is-invalid': error }, className)}
    {...props}
  />
})

Input.defaultProps = {
  type: 'text',
  className: null,
  error: false,
}

Input.propTypes = {
  /** Input valid types (texarea return texarea html tag) */
  type: PT.oneOf([
    'date',
    'datetime-local',
    'email',
    'month',
    'number',
    'password',
    'search',
    'tel',
    'text',
    'textarea',
    'time',
    'url',
    'week',
  ]),
  /** Extra className */
  className: PT.string,
  /** Input is invalid */
  error: PT.bool,
}

Input.displayName = 'Input'

export default Input
