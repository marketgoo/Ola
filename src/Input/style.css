.ola_input {
    --border: var(--color-neutral-400);
    --color: var(--color-neutral-900);
    --background: var(--color-white-100);
    --focus: var(--color-primary-100);

    display: block;
    box-sizing: border-box;
    height: var(--size-5);
    width: 100%;
    border: solid 2px var(--border);
    background: var(--color-white-100);
    color: var(--color);
    background-color: var(--background);
    border-radius: var(--radius-s);
    font: var(--font-1-regular);
    padding: 0 1em;
    margin: 0;
    transition-property: background-color, color, box-shadow, border-color;
    transition-duration: var(--duration);
    outline: 0;
    text-decoration: none;
    font-weight: var(--font-weight-regular);

    &::placeholder {
        color: var(--color-neutral-400);
        opacity: 1;
    }

    &:focus {
        --border: var(--color-primary-500);
        box-shadow: 0 0 0 4px var(--focus);
    }

    &:hover {
        --border: var(--color-primary-600);
    }

    &:invalid,
    &.is-invalid {
        --border: var(--color-negative-500);
        --color: var(--color-negative-500);
        box-shadow: none;

        &:focus {
            box-shadow: 0 0 0 4px var(--color-negative-100);
        }

        &:hover {
            --border: var(--color-negative-600);
        }
    }

    &:disabled {
        --border: var(--color-neutral-200);
        --color: var(--color-neutral-400);
        cursor: not-allowed;

        &::placeholder {
            opacity: 0;
        }
    }

    &[readonly] {
        box-shadow: none;
    }
}

textarea.ola_input {
    padding-top: 1em;
    padding-bottom: 1em;
    height: auto;
    resize: vertical;
}