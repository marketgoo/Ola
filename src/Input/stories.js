import React from 'react'
import Input from './'

export default {
  title: 'Input',
  component: Input,
  args: {
    disabled: false,
    readOnly: false,
    error: false,
    placeholder: 'Text placeholder',
    className: '',
    type: 'text',
    value: ''
  },
  argTypes: {
    onClick: { action: 'clicked' },
    onFocus: { action: 'focused' },
    onBlur: { action: 'blurred' },
    onInput: { action: 'input' },
  },
}

export const InputBase = (args) => <Input {...args} />

export const InputError = (args) => <Input {...args} />
InputError.args = {
  error: true,
}

export const InputDisabled = (args) => <Input {...args} />
InputDisabled.args = {
  disabled: true,
  value: 'Input disabled'
}

export const InputReadOnly = (args) => <Input {...args} />
InputReadOnly.args = {
  readOnly: true,
  value: 'Input readonly'
}

export const TextareaBase = (args) => <Input {...args} />
TextareaBase.args = {
  placeholder: 'Textarea placeholder',
  type: 'textarea',
}

export const TextareaError = (args) => <Input {...args} />
TextareaError.args = {
  placeholder: 'Textarea placeholder',
  type: 'textarea',
  error: true,
}

export const TextareaDisabled = (args) => <Input {...args} />
TextareaDisabled.args = {
  placeholder: 'Textarea placeholder',
  type: 'textarea',
  disabled: true,
  value: 'Textarea disabled'
}

export const TextareaReadOnly = (args) => <Input {...args} />
TextareaReadOnly.args = {
  placeholder: 'Textarea placeholder',
  type: 'textarea',
  readOnly: true,
  value: 'Textarea readonly'
}