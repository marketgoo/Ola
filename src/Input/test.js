import React from 'react'
import Input from './'
import renderer from 'react-test-renderer'

it('Default Input', () => {
  const tree = renderer
    .create(
      <Input placeholder="The placeholder" />
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
