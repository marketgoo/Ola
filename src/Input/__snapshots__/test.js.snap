// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Input 1`] = `
<input
  className="ola_input"
  placeholder="The placeholder"
  type="text"
/>
`;
